tempto-core/src/main/java/io/prestosql/tempto/query/JdbcConnectionsPool.java

```diff
@@ -35,6 +35,9 @@ public Connection connectionFor(JdbcConnectivityParamsState jdbcParamsState)
         }
 
         Connection connection = dataSources.get(jdbcParamsState).getConnection();
+        if (connection == null) {
+            throw new IllegalStateException("No connection was created for: " + jdbcParamsState.getName());
```

Comment:
maybe a comment:

```suggestion
            // this should never happen, `javax.sql.DataSource#getConnection()` should not return null
            throw new IllegalStateException("No connection was created for: " + jdbcParamsState.getName());
```

?